### PR TITLE
Fix create team, update team missing units

### LINKED ISSUE #5187: When creating a team with the api, that team does not have access to anything
- Gitea version (or commit ref): aeb5655
- Operating system:
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
  - [x] not relevant
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes
  - [ ] No
  - [ ] Not relevant
- Log gist:

## Description

When we migrated to Gitea from a bunch of Git repos behind an Apache, we wrote some scripts to move all repos and create users and teams and so on. This worked fairly well, except those users don't have access to the repos (404). 
When I debugged this locally, I discovered that no `team_units` are created when creating a repo using the api, which is why the users won't have access to the repo, whereas when doing this with the web ui, all units are created accordingly.

More specifically:

##### API:

```log
2018/10/25 22:33:47 [I] [SQL] BEGIN TRANSACTION
2018/10/25 22:33:47 [I] [SQL] INSERT INTO `team` (`org_id`,`lower_name`,`name`,`description`,`authorize`,`num_repos`,`num_members`) VALUES (?, ?, ?, ?, ?, ?, ?) []interface {}{2, "testapi", "testapi", "string", 2, 0, 0}
2018/10/25 22:33:47 [I] [SQL] UPDATE `user` SET num_teams=num_teams+1 WHERE id = ? []interface {}{2}
2018/10/25 22:33:47 [I] [SQL] COMMIT
```

###### UI:

```log
2018/10/25 22:34:34 [I] [SQL] BEGIN TRANSACTION
2018/10/25 22:34:34 [I] [SQL] INSERT INTO `team` (`org_id`,`lower_name`,`name`,`description`,`authorize`,`num_repos`,`num_members`) VALUES (?, ?, ?, ?, ?, ?, ?) []interface {}{2, "testui", "testui", "", 2, 0, 0}
2018/10/25 22:34:34 [I] [SQL] INSERT INTO `team_unit` (`org_id`, `team_id`, `type`) VALUES (?, ?, ?),(?, ?, ?),(?, ?, ?),(?, ?, ?),(?, ?, ?),(?, ?, ?),(?, ?, ?) []interface {}{2, 7, 1, 2, 7, 2, 2, 7, 3, 2, 7, 4, 2, 7, 5, 2, 7, 6, 2, 7, 7}
2018/10/25 22:34:34 [I] [SQL] UPDATE `user` SET num_teams=num_teams+1 WHERE id = ? []interface {}{2}
2018/10/25 22:34:34 [I] [SQL] COMMIT
```

## Steps to reproduce

1. Create a team with write access using the api
2. Give that team access to a private repo
2. Add a user (either also via the api or with the gui)
3. Try accessing the repo with the user you added to the team earlier

## Possible solutions

(relevant code is in [models/org_team.go:285](https://github.com/go-gitea/gitea/blob/master/models/org_team.go#L285)

* Always give a team rights to everything (code, issues, pull requests, releases, wiki, external wiki, external issues) -> Fastest, but quick and dirty
* Add an extra parameter to the api to be able to specify what exact permissions the team should have.

### PR DESCRIPTION
**Pull request by lunny**
Should fix #5187. Now when you use the API, you have to give the units on the form if the team is not an admin team. The units is an array of string, it could be `repo.code,repo.issues,repo.ext_issues,repo.wiki,repo.pulls,repo.releases,repo.ext_wiki` now.

### CODE REVIEW

File: models/unit.go [hunk 1]
```diff
@@ -137,3 +141,13 @@ var (
 		UnitTypeExternalWiki:    UnitExternalWiki,
 	}
 )
+
+// FindUnitTypes give the unit key name and return unit
+func FindUnitTypes(nameKeys ...string) (res []UnitType) {
+	for t, u := range Units {
+		if strings.EqualFold(u.NameKey, u.NameKey) {
```
Comment (kolaente) [inline models/unit.go]: This is always `true`. You are comparing `u.NameKey` with `u.NameKey`...

**Review by kolaente — CHANGES_REQUESTED** [review]
Please update swagger.

**Comment by kolaente** [comment]
Should we backport this?

File: models/unit.go (hunk 1 above)
Comment (lunny) [reply models/unit.go]: done.

**Comment by lunny** [comment]
@kolaente I think yes, since v1.6 stable is not released.

**Comment by kolaente** [comment]
CI still fails, looks like swagger is not right.

File: templates/swagger/v1_json.tmpl [hunk 2]
```diff
@@ -21,7 +21,7 @@
     },
     "version": "1.1.1"
   },
-  "basePath": "{{AppSubUrl}}/api/v1",
```
Comment (techknowlogick) [inline templates/swagger/v1_json.tmpl]: @lunny the change on this line needs to be reverted for CI to be successful.

**Review by kolaente — APPROVED** [review]

**Comment by lunny** [comment]
blocked by https://github.com/go-gitea/go-sdk/pull/127

**Comment by lunny** [comment]
@techknowlogick CI PASS

**Review by appleboy — APPROVED** [review]

**Comment by lafriks** [comment]
This wont be breaking change?

**Pull request merged**